client/src/components/ImageSelectModal.js

```diff
@@ -23,12 +23,21 @@ export default class ImageSelectModal extends Component {
         this.selectFile = this.selectFile.bind(this)
         this.submit = this.submit.bind(this)
         this.fileTypeIsImage = this.fileTypeIsImage.bind(this)
+        this.checkFileSizeLimit = this.checkFileSizeLimit.bind(this)
     }
 
     fileTypeIsImage (file) {
         return file && (file.type === 'image/jpeg' || file.type === 'image/jpg' || file.type === 'image/png')
     }
 
+    checkFileSizeLimit(file) {
+        if (file.size > 5 * 1024 * 1024) {
```

Comment:
Let's have a top level constant FILE_SIZE_LIMIT const at top of file.
Also since we're return a boolean, we could simply `return file.size > FILE_SIZE_LIMIT`